pkg/virtctl/imageupload/imageupload.go

```diff
@@ -304,6 +311,7 @@ func (c *command) run(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
+	dataUploaded = true
```

Comment:
I am not sure what exactly in uploadData is, but in case of an error, would we not have the case where `dataUpload` is still false and we would still have a double-close?

Reply:
To tell you the truth I see no real sense using `util.CloseIOAndCheckErr()` for closing a file which was opened only for read, and ding gymnastics to prevent the double-close error.

I would simply fix it back to be to the good old `defer file.Close()`. What do you say?

Reply:
Works for me too, but adding @ezrasilvera who wants to have this for gosec.

Reply:
Independent of that, it is nice to have a defined behaviour on uploadData (like, it won't close the stream).

Reply:
Looking into it

Reply:
Removing this dumb patch, doing it the much better way you suggested.